### ISSUE TITLE
How to Run without exiting on error?

### ISSUE THREAD

**Issue by ebuchman** [urfave/cli#565]
Currently it appears that if a command returns an error, `Run` will exit the program, since it calls HandleExitCoder, which always exits if the error is not nil. Is it possible to receive that error without exiting? For instance, I've implemented an interactive console that uses this library - the console should not exit just because a command returned an error

**Comment by jszwedko** [comment]
@ebuchman agreed, we should support this use case. For now you can workaround it by setting `cli.OsExiter = func(_ int) {} ` to avoid exiting on error, but I'll mark this as an enhancement.

**Comment by wttw** [comment]
Perviously app.Run() would exit only if the command opted in to it by returning an error that implemented ExitCoder. Otherwise it would return that error to the caller for it to handle.

Now HandleExitCoder always exits if err != nil, meaning that app.Run() only ever returns nil.

It looks like an API change (added in f41708) and probably a bug to me, rather than something that needs an enhancement or hack to work around.

**Comment by jszwedko** [comment]
@wttw yeah, I took a deeper dive in and tend to agree with you that the original intent was to only exit if it satisfied `ExitCoder` (in #361 -- cc/ @meatballhat). It was later adjusted in #496 to satisfy #475, but, in hindsight, the recommendation there should have been to use `RunAndExitOnError()`. `RunAndExitOnError` was deprecated when `ExitCoder` was added, but I could see bringing it back as a convenience function for users who want that behavior, but don't want to wrap their errors in `ExitCoder`s.

Long story short, I agree that this is a bug. We'll need to be careful about releasing this though, since some people may have started to depend on this behavior.

**Comment by jszwedko** [comment]
This is fixed by #595 -- thanks for raising it @ebuchman
